### DISCUSSION TITLE
Translator does not translate localised dates

### DISCUSSION THREAD

**Discussion by kochen** [symfony/symfony#59170] [Q&A] [answered]
### Symfony version(s) affected

7.2

### Description

When using the translator to localise dates, the date remain un-localised (i,e uses the locale from the server)

But, twig renders the dates perfectly fine.

### Shared Configuration
```
framework:
    default_locale: es-ES
    translator:
        default_path: '%kernel.project_dir%/translations'
        fallbacks:
            - es-ES
        providers:
```

#### Controller
```php
        $content = $this->translator->trans(
            'today',
            [
                'a_date' => (new \DateTime('2023-01-25 11:00:00'))->format('j F Y'),
            ]
        );
```
#### Result
<img width="397" alt="image" src="https://github.com/user-attachments/assets/bc460b20-b167-4b09-8f91-6721e5d63e88">

#### Possible cause
This probably happens because the date is passed as text, in a specific format, but it cannot be passed as a DateTime:
<img width="929" alt="image" src="https://github.com/user-attachments/assets/b2ee2b3b-e361-45a3-82c0-a82d8203ebff">

#### Controller
```php
    #[Route('/twig', name: 'app_date_twig')]
    public function __invoke(): Response
    {
        $content = $this->twig->render(
            'date.html.twig',
            [
                'a_date' => new \DateTime('2023-01-25 11:00:00')
            ]
        );

        return new Response($content);
    }
```
#### Twig
```
{% extends 'base.html.twig' %}

{% block body %}
    {% trans with {'a_date': a_date|format_date('long')} %}today{% endtrans %}
{% endblock %}
```
#### Translation
```yaml
today: hoy es el {a_date}
```

#### Result
<img width="392" alt="image" src="https://github.com/user-attachments/assets/702db40d-d0f6-4ed6-bfec-aec069738c8a">

**Comment by xabbuh** [comment, marked as answer]
Your PHP code is not equivalent to what you have in your Twig template. The `format_date()` function of the `twig/intl-extra` package uses the `IntlDateFormatter` under the hood. If you do the same in your PHP code, you will receive the same result.

**Reply by kochen** [reply]
> Your PHP code is not equivalent to what you have in your Twig template. The `format_date()` function of the `twig/intl-extra` package uses the `IntlDateFormatter` under the hood. If you do the same in your PHP code, you will receive the same result.

you were right, thank you!
bedankt!
gracias!
grazie!
